index.bs

```diff
@@ -696,11 +922,28 @@ partial interface NeuralNetworkContext {
             with the value -1 is computed so that the total size remains
             constant.
 
-    **Returns:**: an {{Operand}}. The output tensor. The values of the output
+    **Returns:** an {{Operand}}. The output tensor. The values of the output
     tensor are the same as values of the input tensor. The shape of the output
     tensor is specified by the *newShape* argument.
 </div>
 
+### slice ### {#api-neuralnetworkcontext-slice}
+Produce a slice of the input tensor.
+<script type=idl>
+partial interface NeuralNetworkContext {
+  Operand slice(Operand input, sequence<long> starts, sequence<long> ends);
+};
+</script>
+<div algorithm=slice>
+    **Arguments:**
+        - *input*: an {{Operand}}. The input tensor.
+        - *starts*: a sequence of {{long}}. The starting indices of the corresponding axes of the input shape.
+        - *ends*: a sequence of {{long}}. The ending indices of the corresponding axes of the input shape.
+            The ending index value of -1 selects all the remaining tensor values from the starting index of the given axis. 
```

Comment:
If we want to allow -1, it may be convenient to allow any negative value, interpreted as counting back from the end, and to allow it for starts as well.

Reply:
@gramalingam I think we're going to have to choose which semantic we want here, the [numpy](https://numpy.org/doc/stable/reference/arrays.indexing.html#basic-slicing-and-indexing)'s one or the [TensorFlow](https://js.tensorflow.org/api/latest/#slice)'s one. ONNX follows numpy's convention, which is what you suggested.

The TensorFlow's convention is `(start, length)` where `start` is 0-based and `length` is 1-based. Here `length = -1` means the remaining number of element to the end of the axis dimension.

The numpy's convention is `(start, end)` where both are 0-based indices, and both are allowed negative values. Here `end = -1` means `end = length - 1`, so a slice of `(0, -1)` selects all the elements in an axis except the last one. The drawback of this convention is that to select to the end of the axis dimension without knowing the dimension size (length *n*), one would need to pass an artificially big number like INT_MAX i.e. `(0, INT_MAX)`. That's fine but we now need to agree on what that big number maybe e.g. 32 or 64 bit, signed or unsigned, etc. Numpy has a benefit of a Python's missing argument idiom, so one could simply do `(0,)` as absence implies length *n*.

The `(0,-1)` use case of the TensorFlow's convention is quite handy and is used a lot in practice. I'm leaning more toward that convention. Note that PyTorch `narrow` operator follows the `(start, length)` convention as well but the spec does not explicitly state that it supports passing -1 as a size.

Reply:
I agree using -1 to go until the end is convenient. If we treat `(start,end)` as being inclusive of `end`, we could support both features. But it could have the disadvantage of causing some confusion if people are used to particular interpretations.

Reply:
Either way: if the convention is different from ONNX, it will complicate translating between the two. Sigh! I guess we can't make it work for both Tensorflow and ONNX. A compromise is unavoidable. So, I am okay with the proposal